bumble/device.py

```diff
@@ -1172,8 +1172,17 @@ def get_services_by_uuid(self, uuid: core.UUID) -> List[gatt_client.ServiceProxy
         return self.gatt_client.get_services_by_uuid(uuid)
 
     def get_characteristics_by_uuid(
-        self, uuid: core.UUID, service: Optional[gatt_client.ServiceProxy] = None
+        self,
+        uuid: core.UUID,
+        service: Optional[gatt_client.ServiceProxy | core.UUID] = None,
     ) -> List[gatt_client.CharacteristicProxy]:
+        if isinstance(service, core.UUID):
+            services = self.get_services_by_uuid(service)
+            characteristics = []
+            for service in services:
+                characteristics.extend(self.get_characteristics_by_uuid(uuid, service))
+            return characteristics
```

Comment:
suggestion: use a comprehension: `return [self.get_characteristics_by_uuid(uuid, s) for s in self.get_services_by_uuid(service)]`, or a map: `return list(self.get_services_by_uuid(service).map(functools.partial(self.get_characteristics_by_uuid(uuid))))`